homeassistant/components/zone/__init__.py

```diff
@@ -53,11 +63,17 @@
 ICON_HOME = "mdi:home"
 ICON_IMPORT = "mdi:import"
 
+ALLOWED_TYPES = [DEFAULT_TYPE, "polygon"]
+
 CREATE_FIELDS = {
     vol.Required(CONF_NAME): cv.string,
     vol.Required(CONF_LATITUDE): cv.latitude,
     vol.Required(CONF_LONGITUDE): cv.longitude,
+    vol.Optional(ATTR_TYPE, default=DEFAULT_TYPE): vol.In(ALLOWED_TYPES),
```

Comment:
This change requires an architecture discussion.

https://developers.home-assistant.io/docs/core/entity#changing-the-entity-model

Reply:
I'll create a discussion then. Let's leave this as a draft until a decision is made.

Reply:
Let's close it until it is approved.